src/zcash/Address.hpp

```diff
@@ -127,14 +130,85 @@ class UnifiedAddress {
     }
 };
 
+class UnifiedFullViewingKeyBuilder;
+
+/**
+ * Wrapper for a zcash_address::unified::Ufvk.
+ */
+class UnifiedFullViewingKey {
+private:
+    std::unique_ptr<UnifiedFullViewingKeyPtr, decltype(&unified_full_viewing_key_free)> inner;
+
+    UnifiedFullViewingKey() :
+        inner(nullptr, unified_full_viewing_key_free) {}
+
+    UnifiedFullViewingKey(UnifiedFullViewingKeyPtr* ptr) :
+        inner(ptr, unified_full_viewing_key_free) {}
+
+    friend class UnifiedFullViewingKeyBuilder;
+public:
+    static std::optional<UnifiedFullViewingKey> Decode(
+            const std::string& str,
+            const KeyConstants& keyConstants);
+
+    static UnifiedFullViewingKey FromZcashdUFVK(const ZcashdUnifiedFullViewingKey&);
+
+    std::string Encode(const KeyConstants& keyConstants) const;
+
+    std::optional<SaplingDiversifiableFullViewingKey> GetSaplingKey() const;
+
+    std::optional<CChainablePubKey> GetTransparentKey() const;
+
+    UnifiedFullViewingKey(const UnifiedFullViewingKey& key) :
+        inner(unified_full_viewing_key_clone(key.inner.get()), unified_full_viewing_key_free) {}
```

Comment:
The move constructor also needs to exist:
```suggestion
    UnifiedFullViewingKey(UnifiedFullViewingKey&& key) : inner(std::move(bundle.inner)) {}

    UnifiedFullViewingKey(const UnifiedFullViewingKey& key) :
        inner(unified_full_viewing_key_clone(key.inner.get()), unified_full_viewing_key_free) {}
```